src/session_impl.cpp

```diff
@@ -1697,7 +1708,6 @@ namespace aux {
 #ifndef TORRENT_NO_DEPRECATE
 	void session_impl::set_settings(libtorrent::session_settings const& s)
 	{
-		INVARIANT_CHECK;
```

Comment:
I believe this is an external function, isn't it? I think this is accessible via ``session_handle`` and can be called directly by users.